src/test/java/br/com/alura/comex/relatorio/RelatorioVendasCategoriaTest.java

```diff
@@ -0,0 +1,143 @@
+package br.com.alura.comex.relatorio;
+
+import br.com.alura.comex.model.Pedido;
+import br.com.alura.comex.model.builder.PedidoBuilder;
+import br.com.alura.comex.utils.FormatosImpressao;
+import org.junit.jupiter.api.*;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+public class RelatorioVendasCategoriaTest {
+
+    private GeradorRelatorio relatorio;
+    private List<Pedido> listaPedidos;
+
+
+    @BeforeEach
+    public void setup() {
+        relatorio = new RelatorioVendasCategoria();
+    }
+
+    @Test
+    @DisplayName("Teste Relatório Vendas por Categoria - sem lista de pedidos")
+    public void relatorioSemListaDePedidos() {
+
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            relatorio.gerarConteudo(new ArrayList<>());
+        });
+
+    }
+
+    @Test
+    @DisplayName("Teste Relatório Vendas por Categoria - com um pedido ( uma categoria)")
+    public void relatorioDeUmUnicoPedido() {
+
+        listaPedidos = new ArrayList<>();
+        listaPedidos.add(geraListaComUmPedido());
+        assertEquals(relatorio.gerarConteudo(listaPedidos), this.conteudoEsperadoUmUnicoPedido());
+    }
+
+    @Test
+    @DisplayName("Teste Relatório Vendas por Categoria - Pedidos com produtos de varias categorias")
+    public void relatorioComMaisDeUmaCategoria() {
+        assertEquals(relatorio.gerarConteudo(geraListaPedidosComMaisDeUmaCategorias()), conteudoEsperadoMaisDeUmaCategoria());
```

Comment:
Aqui é invertido: primeiro o expected depois o actual!

Eu confesso que acho mais expressivo ter esse código em várias linhas:

```suggestion
        List<Pedido> pedidosComMaisDeUmaCategoria = geraListaPedidosComMaisDeUmaCategorias();
        String conteudoRelatorio = relatorio.gerarConteudo();
        assertEquals(conteudoEsperadoMaisDeUmaCategoria(), conteudoRelatorio);
```